src/gluonts/nursery/autogluon_for_single_ts.py

```diff
@@ -0,0 +1,131 @@
+        return task.Dataset(df[:-2*len]), task.Dataset(df[-2*len:-len]), task.Dataset(df[-len:])
+
+    def change_test(self):
+        for data_entry in self.convert_df(iter([list(self.dataset.test)[0]])):
+            yield data_entry["ts"]
+
+    def convert_df(self, data_iterator):
+        for data_entry in data_iterator:
+            data = data_entry.copy()
+            index = pd.date_range(
+                start=data["start"],
+                freq=self.dataset.metadata.freq,
+                periods=data["target"].shape[-1],
+            )
+            data["ts"] = pd.DataFrame(
+                index=index, data=data["target"].transpose()
+            )
+            yield data
+
+    def to_sampleforecast(self, y_pred):
+        l = 1
+        freq = self.dataset.metadata.freq
+        pred_len = self.dataset.metadata.prediction_length
+        test = [list(self.dataset.test)[0]]
+        forecasts = []
+        for i in range(l):
+            t = test[i]
+            start = to_pandas(t).index[-pred_len]
+            y_hat = np.array(y_pred[i * pred_len:(i + 1) * pred_len])
+            samples = np.vstack([y_hat] * self.num_samples)
+            # print('stacked samples:', samples)
+            sample = SampleForecast(freq=freq, start_date=pd.Timestamp(start, freq=freq), item_id=str(i), samples=samples)
+            forecasts.append(sample)
+        return forecasts
+
+    def evaluate_it(self, predictor, test_data):
+        label_column = 'target'
+        test_data_nolab = test_data.drop(labels=[label_column], axis=1)
+        y_pred = predictor.predict(test_data_nolab)
+        evaluator = Evaluator(quantiles=[0.5])
+        # convert the test dataset
+        ts_it = self.change_test()
+        tss = list(ts_it)
+        # convert the forecasts
+        forecasts = self.to_sampleforecast(y_pred)  # y_pred
+        agg_metrics, item_metrics = evaluator(iter([tss[0][-self.dataset.metadata.prediction_length:]]), iter(forecasts), num_series=1)
+        print(y_pred)
+        # print(tss.values[-5*self.dataset.metadata.prediction_length:])
+        return json.dumps(agg_metrics, indent=4)
+
+    def Wquantile(self, y, y_pred):
+        assert len(y) == len(y_pred)
+        res = 0
+        for i in range(len(y)):
+            res += abs(y.values[i][0]-y_pred[i])
+        return res / sum(abs(y0) for y0 in y_pred)
+
+
+
+g = Gluonts3Auto('m4_weekly', 'regression', 1, context_length=None)
+ds = get_dataset('m4_weekly', regenerate=False)
+test = list(ds.test)[0]
+series = to_pandas(test)
+df = g.parse_series(series)
+task_tr, task_val, task_ts = g.split_df(df)
+label_column = 'target'
+problem_type = 'regression'
+predictor = task.fit(train_data=task_tr, tuning_data=task_val, label=label_column, problem_type=problem_type,
+                     output_directory='AutogluonModels/ag-a') #, presets='best_quality'
+# print(predictor.fit_summary())
+print(g.evaluate_it(predictor, task_ts))
```

Comment:
I think this should all be wrapped into a new GluonTS model called `AutoGluonPredictor` or something like that. Users should be able to use this new GluonTS model just like they can use any other GluonTS models. They should also be able to pass in any kwargs they wish into AutoGluon's task.fit() call.

@lovvge @dcmaddix  Please chime in here on the best way to achieve an integrated API!

Reply:
your class can be called `AutogluonEstimator` and should inherit `Estimator` class:

https://github.com/awslabs/gluon-ts/blob/13eb1f5396240232ddc6ea04ad37eb8fbf3af150/src/gluonts/model/estimator.py#L37

Reply:
I think one could do both:
1. for training on a single time series "on the fly" and directly getting predictions back, the functionality can be wrapped into a class that extends `Predictor` from GluonTS
2. if we want to also have offline training (i.e. train on a bunch of time series and get a regression model to be used in the future for predictions) then that should be done by extending `Estimator`

I think option 1. is simpler to start with